kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -67,115 +65,89 @@ public final class VirtualKafkaClusterReconciler implements
-            var unresolvedFilters = existingFilters.stream()
-                    .filter(kpf -> hasAnyResolvedRefsFalse(Optional.ofNullable(kpf.getStatus()).map(KafkaProtocolFilterStatus::getConditions).orElse(List.of())))
-                    .map(ResourcesUtil::toLocalRef)
-                    .collect(Collectors.toCollection(TreeSet::new));
-
-            TreeSet<ProxyRef> ingressProxyRefs = existingIngresses.stream().map(i -> i.getSpec().getProxyRef()).collect(Collectors.toCollection(TreeSet::new));
-            // if the ingress references a different proxy to the virtual cluster, we will not have loaded the proxy.
-            // TODO can we discriminate between unresolved this mismatch?
-            TreeSet<ProxyRef> mismatchedIngressProxyRef = ingressProxyRefs.stream().filter(i -> !existingProxyRefs.contains(i))
-                    .collect(Collectors.toCollection(TreeSet::new));
-
-            if (unresolvedServices.isEmpty()
-                    && unresolvedIngresses.isEmpty()
-                    && unresolvedFilters.isEmpty()
-                    && mismatchedIngressProxyRef.isEmpty()) {
-                uc = context
-                        .getSecondaryResource(ConfigMap.class, PROXY_CONFIG_MAP_EVENT_SOURCE_NAME)
-                        .flatMap(cm -> Optional.ofNullable(cm.getData()))
-                        .map(ProxyConfigData::new)
-                        .flatMap(data -> data.getStatusPatchForCluster(ResourcesUtil.name(cluster)))
-                        .map(patch -> {
-                            var rr = ResourceState.of(statusFactory.newTrueCondition(cluster, Condition.Type.ResolvedRefs));
-                            var acc = ResourceState.fromList(patch.getStatus().getConditions());
-                            return statusFactory.clusterStatusPatch(cluster, rr.replacementFor(acc));
-                        })
-                        .map(UpdateControl::patchStatus)
-                        .orElse(UpdateControl.noUpdate());
-
-            }
-            else {
-                Stream<String> serviceMsg = refsMessage("spec.targetKafkaServiceRef references ", cluster, unresolvedServices);
-                Stream<String> ingressMsg = refsMessage("spec.ingressRefs references ", cluster, unresolvedIngresses);
-                Stream<String> filterMsg = refsMessage("spec.filterRefs references ", cluster, unresolvedFilters);
-                Stream<String> ingressProxyMessage = refsMessage("a spec.ingressRef had an inconsistent or missing proxyRef ", cluster, mismatchedIngressProxyRef);
-                uc = UpdateControl
-                        .patchStatus(statusFactory.newFalseConditionStatusPatch(cluster, Condition.Type.ResolvedRefs, Condition.REASON_TRANSITIVE_REFS_NOT_FOUND,
-                                joiningMessages(serviceMsg, ingressMsg, filterMsg, ingressProxyMessage)));
-            }
+        ResolutionResult resolutionResult = resolver.deepResolveCluster(cluster, context);
```

Comment:
Not convinced by `deepResolveCluster` How about `resolver.resolveTransitiveRefs(cluster, context)`?

Reply:
we already have the `transitive` word knocking around in the `ResolvedRefs` reason `TransitivelyReferencedResourcesNotFound` which is used when one of the dependencies has a `ResolvedRefs: False`  condition. As opposed to `ReferencedResourcesNotFound` where we cannot find a dependency the VKC refers to.

This resolver is looking at both types of dependencies, direct references from this VKC and then transitive dependencies.

It's resolving direct and transitive dependencies. Maybe `resolveRefs` is better. Definitely think you're right that you don't resolve a cluster, you resolve it's dependencies.